### ISSUE TITLE
is it a way to easily to iteratively chat with the LLM and record the output?

### ISSUE THREAD

**Issue by zyzisastudyreallyhardguy** [h2oai/h2ogpt#440]
Dear h2ogpt team,

I am testing the ability of various LLM. so i want to give a list of strings and obtain responses from LLMs and record them.

Is it any way to easily achieve this? Also, does the framework support multiple GPU for data parrallelism. So that i can get the inference results faster with the help of multiple GPUs?

Best Regards

**Comment by pseudotensor** [comment]
Yes, you can use the "eval" mode for this.  i.e. you would make a json file and use like the tests/test_eval.py::test_eval_json.py runs, passing the same arguments on the CLI or in python like shown.

https://github.com/h2oai/h2ogpt/blob/9e0e35286d6ae022ae41e46659b3786e95a11f11/tests/test_eval.py#L19-L42

This saves the information in parquet in `eval_out_filename` as well as a figure showing scores if you don't pass score_model=None.

**Comment by zyzisastudyreallyhardguy** [comment]
Could you please give me some code examples? I found it still hard to follow. Thanks!

**Comment by pseudotensor** [comment]
That test does exactly the thing you are asking about.  When you look at it, can you explain what you want more than that simple example?

**Comment by zyzisastudyreallyhardguy** [comment]
> That test does exactly the thing you are asking about. When you look at it, can you explain what you want more than that simple example?

How can i open the eval mode?  is it just python generate.py --base_model='mymodel' --eval_file_name 'my_file.json', and all done?

**Comment by pseudotensor** [comment]
If you want to run on CLI, yes like:

```
python generate.py --base_model=MYMODEL --eval_filename=MYFILE.json --eval_prompts_only_num=NPROMPTS
```
where NPROMPTS is the number of prompts in the json file to evaluate (can be less than total).
